packages/xsnap/src/xsnap.js

```diff
@@ -121,7 +142,7 @@ export function xsnap(options) {
         xsnapProcess.kill();
         throw new Error('xsnap protocol error: received empty message');
       } else if (message[0] === OK) {
-        return message.subarray(1);
+        return { reply: message.subarray(1), meterStats };
```

Comment:
Wait, shouldn't the `meterStats` be part of the `OK` that comes back from the worker? Why add a separate message for it? Metering should be measured exactly from the beginning to the end of the crank, and `OK` is how the worker signals the end of the crank, so I'm not in favor of a pathway that suggests that the worker might send metering results at other times.

Reply:
> Wait, shouldn't the `meterStats` be part of the `OK` that comes back from the worker?

I thought it was, at a higher level... but I only see it in the manager...
https://github.com/Agoric/agoric-sdk/pull/2384/files#diff-73740b92f3f3365578c6c1492419c760b2ec870fa71a269349faca6bfb55f63aR164

I'm not sure how this works without any changes to the supervisor. hmm.

Reply:
I've improved this now.  Will request a re-review when it's ready.